src/runtime_src/core/common/api/xrt_bo.cpp

```diff
@@ -833,233 +857,264 @@ sync(const std::string& port, xclBOSyncDirection dir, size_t sz, size_t offset)
 xrtBufferHandle
 xrtBOAllocUserPtr(xrtDeviceHandle dhdl, void* userptr, size_t size, xrtBufferFlags flags, xrtMemoryGroup grp)
 {
-  try {
-    auto boh = alloc(get_xcl_device_handle(dhdl), userptr, size, flags, grp);
-    bo_cache[boh.get()] = boh;
-    return boh.get();
-  }
-  catch (const xrt_core::error& ex) {
-    xrt_core::send_exception_message(ex.what());
-    errno = ex.get();
-  }
-  catch (const std::exception& ex) {
-    send_exception_message(ex.what());
-    errno = 0;
-  }
-  return nullptr;
+  return xdp::native::profiling_wrapper(__func__, nullptr,
+  [dhdl, userptr, size, flags, grp]{
+    try {
```

Comment:
Hmm.  I think the try/catch should be outside profiling_wrapper,  what if profiling_wrapper throws?  The point of the try/catch in the C-APIs is to make sure c++ exceptions don't escape to C code.

Reply:
Switched the wrapper to be inside the try and moved the catches up in the C-API functions in the latest push